app/services/salesforce_subscriber.rb

```diff
@@ -0,0 +1,59 @@
+# Restforce uses faye as the underlying implementation for CometD.
+
+require 'restforce'
+require 'faye'
+
+class SalesforceSubscriber
+  attr_reader :client
+  LOG_PATH = "#{Rails.root}/log/#{Rails.env}_salesforce.log"
+  CONTACT_PUSH_TOPIC_NAME = 'ContactChange'
+  # Initialize a client with your username/password/oauth token/etc.
+  def initialize
+    @client = Restforce.new(username: ENV['SALESFORCE_USERNAME'],
+                            password: ENV['SALESFORCE_PASSWORD'],
+                            security_token: ENV['SALESFORCE_SECURITY_TOKEN'],
+                            client_id: ENV['SALESFORCE_CONSUMER_KEY'],
+                            client_secret: ENV['SALESFORCE_CONSUMER_SECRET'],
+                            host: ENV['SALESFORCE_LOGIN_DOMAIN'])
+  end
+
+  def create_contact_push_topic
+    # Create a PushTopic for subscribing to Contact changes.
+    delete_contact_topics
+    contact_topic = @client.create('PushTopic',
+                                   ApiVersion: '48.0',
+                                   Name: CONTACT_PUSH_TOPIC_NAME,
+                                   Description: 'all contact records',
+                                   NotifyForOperations: 'All',
+                                   NotifyForFields: 'All',
+                                   Query: 'select Id, Email, Faculty_Verified__c from Contact')
+
+    PushTopic.create(topic_salesforce_id: contact_topic, topic_name: CONTACT_PUSH_TOPIC_NAME) if contact_topic.present? && contact_topic.is_a?(String)
+    Logger.new(LOG_PATH).debug('Contact Push Topic Id: ' + contact_topic)
+  rescue Restforce::ErrorCode::DuplicateValue
+    delete_contact_topics
+    create_contact_push_topic
+  end
+
+  def subscribe
+    authorization_hash = @client.authenticate!
+    @client.faye.set_header 'Authorization', "OAuth #{authorization_hash.access_token}"
+    EM.run do
+      @client.subscription "/topic/#{CONTACT_PUSH_TOPIC_NAME}", replay: -1 do |message|
+        Logger.new(LOG_PATH).debug('Contact Received')
+        ContactParser.new(message).save_contact
+      end
+    end
```

Comment:
maybe we can move some of this into `openstax_salesforce` and pass in application-specific stuff in a block... 🤔

Reply:
I assume you mean this part?

```
authorization_hash = @client.authenticate!
@client.faye.set_header 'Authorization', "OAuth #{authorization_hash.access_token}"
```

I'll look into it, `openstax_salesforce` doesn't use faye right now.